app/components/post_draft/post_input/post_input.tsx

```diff
@@ -0,0 +1,266 @@
+            placeholder = {id: t('create_comment.addComment'), defaultMessage: 'Add a comment...'};
+        } else {
+            placeholder = {id: t('create_post.write'), defaultMessage: 'Write to {channelDisplayName}'};
+        }
+
+        return placeholder;
+    };
+
+    const checkMessageLength = useCallback((newValue: string) => {
+        const valueLength = newValue.trim().length;
+
+        if (valueLength > maxMessageLength) {
+            // Check if component is already aware message is too long
+            if (!longMessageAlertShown) {
+                Alert.alert(
+                    intl.formatMessage({
+                        id: 'mobile.message_length.title',
+                        defaultMessage: 'Message Length',
+                    }),
+                    intl.formatMessage({
+                        id: 'mobile.message_length.message',
+                        defaultMessage: 'Your current message is too long. Current character count: {count}/{max}',
+                    }, {
+                        max: maxMessageLength,
+                        count: valueLength,
+                    }),
+                );
+                setLongMessageAlertShown(true);
+            }
+        } else if (longMessageAlertShown) {
+            setLongMessageAlertShown(false);
+        }
+    }, [intl, longMessageAlertShown, maxMessageLength]);
+
+    const handlePostDraftSelectionChanged = useCallback((event: NativeSyntheticEvent<TextInputSelectionChangeEventData> | null, fromHandleTextChange = false) => {
+        const cp = fromHandleTextChange ? cursorPosition : event!.nativeEvent.selection.end;
+
+        if (Platform.OS === 'ios') {
+            const newKeyboardType = switchKeyboardForCodeBlocks(value, cp);
+            setKeyboardType(newKeyboardType);
+        }
+
+        updateCursorPosition(cp);
+    }, [updateCursorPosition, cursorPosition]);
+
+    const handleTextChange = useCallback((newValue: string) => {
+        updateValue(newValue);
+        lastNativeValue.current = newValue;
+
+        checkMessageLength(newValue);
+
+        // Workaround to avoid iOS emdash autocorrect in Code Blocks
+        if (Platform.OS === 'ios') {
+            handlePostDraftSelectionChanged(null, true);
+        }
+
+        if (newValue && lastTypingEventSent.current + timeBetweenUserTypingUpdatesMilliseconds < Date.now()) {
+            userTyping(serverUrl, channelId, rootId);
+            lastTypingEventSent.current = Date.now();
+        }
+    }, [updateValue, checkMessageLength, handlePostDraftSelectionChanged, timeBetweenUserTypingUpdatesMilliseconds, channelId, rootId]);
+
+    const onPaste = useCallback(async (error: string | null | undefined, files: PastedFile[]) => {
+        if (error) {
+            showPasteFilesErrorDialog(intl);
+        }
+
+        addFiles(await extractFileInfos(files));
+    }, [addFiles, intl]);
+
+    const handleHardwareEnterPress = useCallback((keyEvent: {pressedKey: string}) => {
```

Comment:
Tried this with the emulator. It does not send the "shift-enter" event. Tricking the system, both behaviours work as they should, but not sure if it is exactly correct.

Reply:
I can give it a test drive tomorrow